quilt/test/test_install.py

```diff
@@ -2,47 +2,78 @@
 Tests for commands.
 """
 
+import hashlib
 import json
 import os
 
 import requests
 import responses
 
 from quilt.tools import command
+from quilt.tools.const import HASH_TYPE, TYPE_KEY, NodeType
+from quilt.tools.hashing import hash_contents
+
 from .utils import QuiltTestCase
 
 class InstallTest(QuiltTestCase):
+    """
+    Unit tests for quilt install.
+    """
     # Note: we're using the deprecated `assertRaisesRegexp` method because
     # the new one, `assertRaisesRegex`, is not present in Python2.
 
     def test_install_latest(self):
-        contents = 'HDF5 package'
-        contents_hash = 'e867010701edc0b1c8be177e02a93aa3cb1342bb1123046e1f6b40e428c6048e'
+        """
+        Install the latest update of a package.
+        """
+        tabledata = "table"*10
+        h = hashlib.new(HASH_TYPE)
+        h.update(tabledata.encode('utf-8'))
+        obj_hash = h.hexdigest()
+        contents = dict(foo={TYPE_KEY: NodeType.GROUP.value,
+                             'bar' : {TYPE_KEY : NodeType.TABLE.value,
+                                      'hashes': [obj_hash]}
+                            })
+        contents_hash = hash_contents(contents)
 
         self._mock_tag('foo/bar', 'latest', contents_hash)
-        self._mock_package('foo/bar', contents_hash)
-        self._mock_s3(contents_hash, contents)
+        self._mock_package('foo/bar', contents_hash, contents, [obj_hash])
+        self._mock_s3(obj_hash, tabledata)
 
         session = requests.Session()
         command.install(session, 'foo/bar')
 
-        with open('quilt_packages/foo/bar.h5') as fd:
-            file_contents = fd.read()
+        with open('quilt_packages/foo/bar.json') as fd:
+            file_contents = json.loads(fd.read())
             assert file_contents == contents
 
+        with open('quilt_packages/objs/{hash}{ext}'.format(hash=obj_hash, ext='.h5')) as fd:
+            file_contents = fd.read()
+            assert file_contents == tabledata
+
     def test_bad_hash(self):
```

Comment:
Actually, here, it's more important to test for bad object hashes than bad content hashes. A bad content hash would be a server bug, and should never happen (at least, that's what server unit tests are for). But a bad object hash can totally happen until we start verifying the contents in S3.